### PR TITLE
[Remove Stream/Optional: Phase 3] LPS-181532 module: headless-delivery-Impl

### PR DESCRIPTION
**Pull request by JanisZhang**
https://issues.liferay.com/browse/LPS-181532

### CODE REVIEW

**Comment by JanisZhang** [comment]
ci:test:headless

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/NavigationMenuResourceImpl.java [hunk 1]
```diff
@@ -682,23 +679,25 @@ private void _updateNavigationMenuItems(
...
-					siteNavigationMenuItems.stream();
-
 				Long navigationMenuItemId = navigationMenuItem.getId();
 
-				Optional<SiteNavigationMenuItem>
-					siteNavigationMenuItemOptional = stream.filter(
-						siteNavigationMenuItem -> Objects.equals(
+				SiteNavigationMenuItem existingSiteNavigationMenuItem = null;
+
+				for (SiteNavigationMenuItem siteNavigationMenuItem :
```
Comment (dantewang) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/NavigationMenuResourceImpl.java:686]: @JanisZhang Should call this one `curSiteNavigationMenuItem`.

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/StructuredContentDTOConverter.java [hunk 2]
```diff
@@ -220,15 +218,17 @@ public StructuredContent toDTO(
 	private Map<Locale, String> _filterDescriptionMap(
 		Map<Locale, String> descriptionMap) {
 
-		Set<Map.Entry<Locale, String>> set = descriptionMap.entrySet();
+		Map<Locale, String> filterDescriptions = new HashMap<>();
```
Comment (dantewang) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/StructuredContentDTOConverter.java:221]: @JanisZhang See https://liferay.atlassian.net/wiki/spaces/ENGR/pages/1511752519/Java+conventions+in+Liferay#%5BinlineExtension%5DMaps-are-plurals%2C-with-suffix for naming.

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by dantewang** [comment]
@JanisZhang Please update the naming

**Comment by dantewang** [comment]
See https://github.com/liferay-core-infra/liferay-portal/pull/4270

**Pull request closed without merging**